banyan/src/forest/read.rs

```diff
@@ -0,0 +1,458 @@
+use super::{
+    BranchCache, Config, CryptoConfig, FilteredChunk, FutureResult, ReadForest, TreeTypes,
+};
+use crate::{
+    index::deserialize_compressed, index::zip_with_offset, index::Branch, index::BranchIndex,
+    index::CompactSeq, index::Index, index::IndexRef, index::Leaf, index::LeafIndex,
+    index::NodeInfo, query::Query, store::ArcReadOnlyStore,
+};
+use anyhow::{anyhow, Result};
+use bitvec::prelude::BitVec;
+use core::fmt::Debug;
+use futures::{prelude::*, stream::BoxStream};
+use salsa20::{stream_cipher::NewStreamCipher, stream_cipher::SyncStreamCipher, XSalsa20};
+use serde::{de::DeserializeOwned, Serialize};
+use std::sync::Arc;
+use tracing::info;
+
+/// basic random access append only async tree
+impl<T, V> ReadForest<T, V>
+where
+    T: TreeTypes + 'static,
+    V: Serialize + DeserializeOwned + Clone + Send + Sync + Debug + 'static,
+{
+    pub(crate) fn crypto_config(&self) -> &CryptoConfig {
+        &self.crypto_config
+    }
+
+    pub(crate) fn config(&self) -> &Config {
+        &self.config
+    }
+
+    pub(crate) fn value_key(&self) -> salsa20::Key {
+        self.crypto_config().value_key
+    }
+
+    pub(crate) fn index_key(&self) -> salsa20::Key {
+        self.crypto_config().index_key
+    }
```

Comment:
not sure I have the full picture of what a forest is: if there is a single salsa key, then one forest must correspond to one ActyxOS StreamId, is that correct? (because each stream needs to be protected with a separate key to make it possible to expose it individually)